utils/validate_release_links.py

```diff
@@ -72,28 +74,47 @@ def get_release_tag() -> Optional[str]:
     raise RuntimeError("Can't determine release tag")
 
 
-def check_file(filename: str, global_allow_pattern: Pattern) -> List[str]:
+def check_file(
+    filename: str, global_allow_pattern: Pattern, release_tag: str
+) -> List[str]:
     """
     Validate a single file and return any offending lines.
     """
     bad_lines = []
-    with open(filename) as f:
-        for line in f:
-            if not RELEASE_PATTERN.search(line):
-                continue
-
-            if global_allow_pattern.search(line):
-                continue
-
-            if filename in ALLOW_LIST:
-                if ALLOW_LIST[filename] is None or ALLOW_LIST[filename].search(line):
-                    continue
+    new_file_name = f"{filename}.orig"
+    with open(new_file_name, "a") as new_file:
+        with open(filename) as f:
+            for line in f:
+                keepLine = True
+                keepLine = not RELEASE_PATTERN.search(line)
+                keepLine |= global_allow_pattern.search(line) is not None
+                if filename in ALLOW_LIST:
```

Comment:
nit: could hoist this out of the `for line` loop. And you could simplify a bit with something like
```python
allow_list_pattern = ALLOW_LIST.get(filename, re.compile(""))
...
keep_line |= allow_list_pattern.search(line) is not None
```
(I think - I'm assuming the empty string `re` matches everything)

Reply:
good idea